editor/styling/stylesheets/modules/_outline.scss

```diff
@@ -76,6 +68,39 @@
       @include colorize-disclosure-arrow(-df-error-severity-fatal-dim);
     }
 
+    .label {
+      -fx-min-width: 40px;
+    }
```

Comment:
This PR enables ellipsis on text-overrun, to allow displaying the toggle at the right side of the cell. That might not be ideal on deeply nested outlines. Although horizontal scrolling is not ideal either, this is a middle ground solution that always shows part of the label, and enables horizontal scrolling sooner. For what is worth, I tried to make the visibility toggle stick to the right, without blocking scroll, but I didn't manage to produce a good result. The main problem was that it required watching the scrolling position to rerender the button on a different anchor point, and I couldn't make this work properly.

Reply:
In many ways it would be nicer if the eye icons were in their own column to the right of the tree view item, but I understand that would necessitate more drastic changes to the view.

For now, I think this is fine. Maybe we'll revisit this later if we need to!